File: CovidWatch/app/src/main/java/com/riskre/covidwatch/data/DateConverter.java
package com.riskre.covidwatch.data;
//  Created by Zsombor SZABO on 17/03/2020.
//  Copyright © IZE. All rights reserved.
//  See LICENSE.txt for licensing information.
//  

import androidx.room.TypeConverter;

import java.util.Date;

public class DateConverter {

    @TypeConverter
    public static Date toDate(Long dateLong){
        return dateLong == null ? null: new Date(dateLong);
    }

    @TypeConverter
    public static Long fromDate(Date date){
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/org/covidwatch/android/data/converter/DateConverter.kt
package org.covidwatch.android.data.converter

import androidx.room.TypeConverter
import java.util.Date

class DateConverter {
    @TypeConverter
    fun toDate(dateLong: Long?) = dateLong?.let { Date(it) }

    @TypeConverter
    fun fromDate(date: Date?) = date?.time
}